topbar.js

```diff
@@ -296,17 +296,9 @@ var WorkspaceMenu = Utils.registerClass(
       this._prefItem.connect("activate", () => {
         this.menu.close(true);
         let wi = workspaceManager.get_active_workspace_index();
-        let env = GLib.get_environ();
-        env.push(`PAPERWM_PREFS_SELECTED_WORKSPACE=${wi}`);
-        try {
-          GLib.spawn_async(
-            null,
-            ["gnome-shell-extension-prefs", "paperwm@hedning:matrix.org"],
-            env,
-            GLib.SpawnFlags.SEARCH_PATH | GLib.SpawnFlags.DO_NOT_REAP_CHILD,
-            null
-          );
-        } catch (e) {}
+        let temp_file = Gio.File.new_for_path(GLib.get_tmp_dir()).get_child('paperwm.workspace')
```

Comment:
Quite ugly, huh? Unfortunately are no more spawned as separate process so previous env trick will not work. Also in `prefs` there is no `global` or `imports.gi.Shell`. File is the best approach I've found